src/ui/pages/Menu/Menu.tsx

```diff
@@ -17,34 +17,52 @@ import {
   fingerPrintOutline,
   idCardOutline,
 } from "ionicons/icons";
+import { useMemo, useState } from "react";
 import { TabLayout } from "../../components/layout/TabLayout";
 import { useAppDispatch } from "../../../store/hooks";
-import {
-  setCurrentOperation,
-  setCurrentRoute,
-} from "../../../store/reducers/stateCache";
+import { setCurrentRoute } from "../../../store/reducers/stateCache";
 import { TabsRoutePath } from "../../../routes/paths";
 import "./Menu.scss";
 import { i18n } from "../../../i18n";
-import { OperationType } from "../../globals/types";
+import { Settings } from "./components/Settings";
+import { SubMenu } from "./components/SubMenu";
+import { SubMenuKey } from "./Menu.types";
+
+const submenuMap = new Map([
+  [
+    SubMenuKey.Settings,
+    {
+      Component: Settings,
+      title: "settings.sections.header",
+      addtionalButtons: <></>,
+    },
+  ],
+]);
 
 const Menu = () => {
   const pageId = "menu-tab";
   const dispatch = useAppDispatch();
+  const [showSubMenu, setShowSubMenu] = useState(false);
+  const [selectedOption, setSelectedOption] = useState(0);
```

Comment:
Ah, this should be `useState<SubMenuKey | undefined>();` right? To use the enum properly. Or is everything auto casting to int